metrics/docker-compose.yml

```
@@ -1,18 +1,48 @@
 services:
   prometheus:
-    network_mode: "host" # -- only linux
     image: prom/prometheus
     container_name: prometheus
-    command:
-      - "--config.file=/etc/prometheus/prometheus.yml"
+    restart: unless-stopped
     ports:
       - 9090:9090
-    restart: unless-stopped
     volumes:
       - ./prometheus:/etc/prometheus
       - prom_data:/prometheus
+    command:
+      - "--config.file=/etc/prometheus/prometheus.yml"
+      # "--log.level=debug" # Not receiving metrics? Consider uncommenting
+
+    # Prometheus needs to be able to connect to the host network as "sig" isn't
+    # containerised. No other service should talk to the host.
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+
+  loki:
+    image: grafana/loki:3.0.0
+    container_name: loki
+    ports:
+      - "3100:3100"
+    volumes:
+      - ./loki:/etc/loki
+    command:
+      - "--config.file=/etc/loki/loki.yml"
+
+  alloy:
+    image: grafana/alloy:v1.3.1
+    container_name: alloy
+    ports:
+      - "3200:3200"
+    volumes:
+      - "./alloy/config.alloy:/etc/alloy/config.alloy"
+      - "../logs:/var/log/alloy"
```

Not very important at this point, but it's a good practice to mount as read-only if you know the files don't need to be modified by the container. Does `config.alloy` need to be modified by the application?
```suggestion
      - "./alloy/config.alloy:/etc/alloy/config.alloy:ro"
      - "../logs:/var/log/alloy:ro"
```

---

addressed